pkg/deploy/rp.go

```diff
@@ -414,10 +418,11 @@ EOF
 
 az login -i --allow-no-subscriptions
 
-az keyvault secret download --file /etc/mdm.pem --id "$RPMDMCERTIFICATEVAULTID"
+SVCVAULTURI="$(az keyvault list -g "$RESOURCEGROUPNAME" --query "[?tags.vault=='service'].properties.vaultUri" -o tsv)"
+az keyvault secret download --file /etc/mdm.pem --id "${SVCVAULTURI}secrets/rp-mdm"
 chmod 0600 /etc/mdm.pem
 
-az keyvault secret download --file /etc/mdsd.pem --id "$RPMDSDCERTIFICATEVAULTID"
+az keyvault secret download --file /etc/mdsd.pem --id "${SVCVAULTURI}secrets/rp-mdsd"
```

Comment:
I don't think this deployment will ever succeed in a new region first time.  It will create the Key Vault then deploy the vmss before we have a reasonable amount of time to upload these two certs. Then these calls will fail.  We could then delete the vmss upload the certs and rerun the deployment.  But with now I count 5 secrets required to be in the service keyvault; before the RP can run in a healthy state. I'm beginning to wonder if we want to move the creation of the key vaults(at least the service but probably should just move both) into a separate deployment step. That way we can prep them before creating the vmss/deploying the rp.

Reply:
I agree.

Reply:
The digicert pre-configuration work is also relevant here.  The answer might be as simple as moving the certs to the "nsg" deployment.

Reply:
@mjudeikis wdyt? ^

Reply:
Yes, agreed, it needs to move.
I was running into the same issues with my tests, `s/-nsg/-predeploy`

Reply:
have updated VSTS deployment story to include separating the keyvaults into some other template.